### PR TITLE
feat: ORV2-2299 Letter Of Authorization : Part 1

### PR DESCRIPTION
**Pull request by praju-aot**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)
ORV2-2209

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manual Test via swagger
- [x] npm run test
- [x] npm run test:e2e 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1500-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1500-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1500-dops.apps.silver.devops.gov.bc.ca/api)
- [Policy](https://onroutebc-1500-policy.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1500-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: vehicles/src/modules/loa/loa.service.ts [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+    companyId: number,
+    file: Express.Multer.File,
+  ): Promise<ReadLoaDto> {
+    const readFileDto = await this.dopsService.upload(
+      currentUser,
+      companyId,
+      file,
+    );
+
+    const loa = await this.classMapper.mapAsync(
```
Comment (praju-aot) [inline vehicles/src/modules/loa/loa.service.ts:66]: @gchauhan-aot Please ensure created user name, guid, updated user name  etc is persisted

File: vehicles/src/modules/loa/loa.service.ts [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+        loa: { loaId: loaId },
+      });
+      const isActive = existingLoaDetail.isActive;
+      const documentId = readFileDto?.documentId;
+      const updatedLoaDetail: LoaDetail = await this.classMapper.mapAsync(
+        updateLoaDto,
+        UpdateLoaDto,
+        LoaDetail,
+        {
+          extraArgs: () => ({
```
Comment (praju-aot) [inline vehicles/src/modules/loa/loa.service.ts:287]: @gchauhan-aot Please ensure created user name, guid, updated user name  etc is persisted

File: vehicles/src/modules/common/dops.service.ts [hunk 3]
```diff
@@ -308,4 +311,56 @@ export class DopsService {
...
+    documentId?: Nullable<string>,
+  ): Promise<ReadFileDto> {
+    // Construct the base URL
+    const baseUrl = new URL(process.env.DOPS_URL);
+    // Append the path and query parameter(s)
+    if (!documentId) {
+      baseUrl.pathname += 'dms/upload';
+    } else {
+      baseUrl.pathname += `dms/upload/${documentId}`;
+    }
```
Comment (krishnan-aot) [inline vehicles/src/modules/common/dops.service.ts]:
```suggestion
    baseUrl.pathname += 'dms/upload';
    if (documentId) {
      baseUrl.pathname += '/${documentId}';
    }
```

File: vehicles/src/modules/loa/loa.service.ts (hunk 1 above)
Comment (gchauhan-aot) [reply vehicles/src/modules/loa/loa.service.ts:66]: done

File: vehicles/src/modules/loa/loa.service.ts (hunk 2 above)
Comment (gchauhan-aot) [reply vehicles/src/modules/loa/loa.service.ts:287]: done

**Review by gchauhan-aot — APPROVED** [review]

**Review by krishnan-aot — APPROVED** [review]

**Pull request merged**